diff --git a/modules/areapicker/AreaPicker.qml b/modules/areapicker/AreaPicker.qml
index 6718878..5933e4c 100644
--- a/modules/areapicker/AreaPicker.qml
+++ b/modules/areapicker/AreaPicker.qml
@@ -18,0 +18,4 @@ Scope {
+        function openRegion(): void {
+            Quickshell.execDetached(["@SCREENSHOT_REGION_COMMAND@"]);
+        }
+
@@ -52,4 +56 @@ Scope {
-            root.freeze = false;
-            root.closing = false;
-            root.clipboardOnly = false;
-            root.activeAsync = true;
+            root.openRegion();
@@ -88,4 +89 @@ Scope {
-            root.freeze = false;
-            root.closing = false;
-            root.clipboardOnly = false;
-            root.activeAsync = true;
+            root.openRegion();
